### PR TITLE
can call analytics for an array of movieIds

### PR DESCRIPTION
**Pull request by Hugo-Hontans**
- [x] Make getMovieAnalytics backend function be usable with an array of movieIds instead of one movieId

### CODE REVIEW

File: apps/backend-functions/src/bigQuery.ts [hunk 1]
```diff
@@ -61,20 +63,25 @@ const aggregateEvents = (events: EventAnalytics[], daysPerRange: number) => {
   };
 };
 
+const mergeMovieIdPageInMovieId = (rows: any[]) => {
+  return rows.map(row => row.movieIdPage ? { ...row, movieId: row.movieIdPage } : row)
```
Comment (lsenta) [inline apps/backend-functions/src/bigQuery.ts]: Sauvage,
remove the `movieIdPage` field from the result or someone, somehow, will end up rely on this, which would be a very bad id.
Add a comment.

```
import {omit} from 'lodash';
```

```
// merge the movieIdPage field from bigquery with the movieId field when relevant.
rows.map(row => omit({...row, movieId: row.movieId || row.movieIdPage, ['movieIdPage']);
```

Comment (lsenta) [reply apps/backend-functions/src/bigQuery.ts]: (missing `}` left as an exercise to the reader)

File: libs/movie/src/lib/movie/+state/movie.firestore.ts [hunk 2]
```diff
@@ -62,7 +62,9 @@ export const enum StoreStatus {
 export interface EventAnalytics {
   event_date: string,
   event_name: AnalyticsEvents,
-  hits: number
+  hits: number,
+  movieId: string,
+  movieIdPage: string
```
Comment (lsenta) [inline libs/movie/src/lib/movie/+state/movie.firestore.ts]: don't, that's a detail of implem.

File: libs/movie/src/lib/movie/+state/movie.service.ts [hunk 3]
```diff
@@ -139,6 +139,6 @@ export class MovieService extends CollectionService<MovieState> {
   /** Call a firebase function to get analytics specify to a movie.*/
   public getMovieAnalytics(): Observable<MovieAnalytics> {
     const f = this.functions.httpsCallable('getMovieAnalytics');
-    return f({ movieId: this.query.getActiveId(), daysPerRange: 28 });
+    return f({ movieId: [this.query.getActiveId()], daysPerRange: 28 });
```
Comment (lsenta) [inline libs/movie/src/lib/movie/+state/movie.service.ts]: Nice, 
do you think we should limit the number of movies a user can ask in a query? (in the backend).

Comment (Hugo-Hontans) [reply libs/movie/src/lib/movie/+state/movie.service.ts]: Normally the number of movies that can be called is limited by the number of movies that the organization can access.

**Pull request merged**